app/src/main/contrib/components/Responsibilities/BurdenEstimates/CreateBurdenEstimateSetForm.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from "react";
+import {responsibilityStore} from "../../../stores/ResponsibilityStore";
+import {OptionSelector} from "../../OptionSelector/OptionSelector";
+import {Form} from "./Form";
+import {BurdenEstimateSetTypeCode} from "../../../../shared/models/Generated";
+import {FormEvent} from "react";
+
+interface BurdenEstimateProps {
+    groupId: string;
+    touchstoneId: string;
+    scenarioId: string;
+}
+
+interface CreateBurdenEstimateSet {
+    type: { type: BurdenEstimateSetTypeCode, details: string }
+}
+
+interface FormData {
+    data: CreateBurdenEstimateSet
+}
+
+export class CreateBurdenEstimateSetForm extends React.Component<BurdenEstimateProps, FormData> {
+
+    constructor() {
+        super();
+        this.state = {
+            data: {
+                type: {
+                    type: null,
+                    details: null
+                }
+            }
+        }
+    }
+
+    static successCallback() {
+        responsibilityStore.refreshResponsibilities()
+    }
+
+    onTypeChange(value: BurdenEstimateSetTypeCode) {
+        const newData = this.state.data;
+        newData.type.type = value;
+        this.setState({
+            data: newData
+        })
+    }
```

Comment:
It might be better to use `CreateBurdenEstimateSet` directly as the state type. Then you can write state changes more succinctly as:

```
onDetailsChange(e: FormEvent<HTMLInputElement>) {
    this.setState((prevState) => {
        type: prevState.type.assign({
             details: e.target.value
        });
    })
}

onTypeChange(value: BurdenEstimateSetTypeCode) {
    this.setState((prevState) => {
        type: prevState.type.assign({
            type: value
        });
    })
}
```

Either way, isn't it better to use the prevState version of setState, or explicitly copy the old state object. It feels here like you are directly modifying the state object, which I think is not the normal React way of writing these functions.